libsolidity/ast/Types.h

```diff
@@ -338,9 +385,26 @@ class IntegerConstantType: public Type
 
 	/// @returns the smallest integer type that can hold the value or an empty pointer if not possible.
 	std::shared_ptr<IntegerType const> integerType() const;
+	/// @returns the smallest fixed type that can hold the value or an empty pointer
+	std::shared_ptr<FixedType const> fixedType() const;
+	short scalingFactor() const { return m_scalingFactor; }
+
+	bool isInteger() const { return m_modifier == Modifier::Integer; }
+	bigint leftOfRadix() const
+	{
+		return m_value / boost::multiprecision::pow(bigint(10), m_scalingFactor);
+	}
+	bigint rightOfRadix() const
+	{
+		return m_value % boost::multiprecision::pow(bigint(10), m_scalingFactor);
+	}
+	short commonScalingFactor(short const& _firstNumber, short const& _secondNumber, Token::Value const& _operator) const;
 
 private:
+
 	bigint m_value;
+	short m_scalingFactor;
```

Comment:
Please add a comment, especially about what happens if we e.g. compute `1.0 / 3`.